NetJSON/NetJSON.cs

```diff
@@ -3727,80 +3727,87 @@ public static object Deserialize(Type type, string value, NetJSONSettings settin
             {
                 return null;
             }
+			ptr += index;
+			var startPtr = ptr;
```

Comment:
Remove the ptr += index;
And use the startPtr for all the arithmetic changes on that pointers to avoid moving the position of the original pointer.